src/components/Card.js

```diff
@@ -12,34 +17,58 @@ export default class Card {
       .content
       .querySelector('.card')
       .cloneNode(true);
-
     return cardElement;
   }
 
+  _renderLikes() {
+    if (this._likeCountElement) {
+      this._likeCountElement.textContent = this._likes.length;
+    }
+  }
+
+  _setEventListeners() {
+    this._cardImage.addEventListener('click', () => {
+      this._handleCardClick(this._name, this._link);
+    });
+
+    this._likeButton.addEventListener('click', () => {
+      const isLiked = this._likeButton.classList.contains('card__like-button_active');
+      this._likeButton.classList.toggle('card__like-button_active'); 
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.

You'll have only this:

```js

    this._likeButton.addEventListener('click', () => {
      const isLiked = this._likeButton.classList.contains('card__like-button_active');
      this._handleLikeClick(this._id, !isLiked, this);
    });
    
```

Reply:
* My remark hasn't been corrected https://github.com/XGLogan/se_project_aroundtheus/pull/1#discussion_r2025637300